query/common/src/main/java/org/codice/ddf/admin/common/fields/base/BaseFunctionField.java

```diff
@@ -13,30 +13,66 @@
  **/
 package org.codice.ddf.admin.common.fields.base;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
 
 import org.codice.ddf.admin.api.DataType;
 import org.codice.ddf.admin.api.fields.FunctionField;
 import org.codice.ddf.admin.api.report.ErrorMessage;
 import org.codice.ddf.admin.api.report.FunctionReport;
 import org.codice.ddf.admin.api.report.Report;
 import org.codice.ddf.admin.common.report.FunctionReportImpl;
+import org.codice.ddf.admin.common.report.message.DefaultMessages;
 
 import com.google.common.collect.ImmutableList;
 
 public abstract class BaseFunctionField<T extends DataType> extends BaseField<Map<String, Object>, FunctionReport<T>> implements FunctionField<T> {
 
     private FunctionReportImpl<T> report;
 
+    private String description;
+
     public BaseFunctionField(String functionName, String description) {
         super(functionName, description);
+        this.description = description;
         report = new FunctionReportImpl<>();
     }
 
+    @Override
+    public String description() {
+        if (this.getErrorCodes() != null) {
+            return String.format("%s The possible errors are:    \r\n %s.",
+                    description,
+                    formatErrorCodes(getErrorCodes()));
+        }
+        return super.description();
+    }
+
     public abstract T performFunction();
 
+    public Set<String> getFunctionErrorCodes() {
+        Set<String> errors = new HashSet<>();
+        errors.add(DefaultMessages.MISSING_REQUIRED_FIELD);
+        return errors;
+    }
+
+    public Set<String> getErrorCodes() {
+        Set<String> allErrorCodes = new HashSet<>();
+        for (DataType arg : getArguments()) {
+            Stream.of(allErrorCodes, arg.getErrorCodes())
```

Comment:
❗️ This is painfully inefficient. You're doing something akin to a map-reduce here with an accumulator, but you're going to force hash collisions on each subsequent call as you re-add the current contents of the `allErrorCodes` collection to itself.

Alternate, more efficient and clearer implementations:
```java
        Set<String> allErrorCodes = new HashSet<>();
        allErrorCodes.addAll(getFunctionErrorCodes());

        allErrorCodes.addAll(getArguments().stream()
                .map((Function<DataType, Set>) Field::getErrorCodes)
                .flatMap(Collection<String>::stream)
                .collect(Collectors.toSet()));
        return allErrorCodes;
```
or
```java
        Set<String> allErrorCodes = new HashSet<>();
        allErrorCodes.addAll(getFunctionErrorCodes());

        for (DataType dataType : getArguments()) {
            allErrorCodes.addAll(dataType.getErrorCodes());
        }
        return allErrorCodes;
```
I recommend the latter as the simpler and more efficient.